### PR TITLE
x11-drivers/nvidia-drivers: changes batch

### PR DESCRIPTION
**Pull request by ionenwks**

### CODE REVIEW

**Comment by ionenwks** [comment]
Added 460.56 self-stabling commit (as discussed) here for convenience.

No known regressions, untouched in 30 days, working good in stable, might help with NULL pointer issues better than current 460.39 does.

460.39 will be removed at a later date, want to keep a delay between stable and drop old for these.

**Comment by ionenwks** [comment]
Added another fix for new [bug #779583](https://bugs.gentoo.org/779583)

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+	sed 's/__USER__/nvpd/' \
+		nvidia-persistenced/init/systemd/nvidia-persistenced.service.template \
+		> nvidia-persistenced.service || die
+
+	gzip -d nvidia-{cuda-mps-control,smi}.1.gz || die
+}
+
+nvidia-drivers_make() {
+	emake -C nvidia-${1} ${2} \
+		DESTDIR="${ED}" PREFIX=/usr LIBDIR="${ED}/usr/$(get_libdir)" \
```
Comment (thesamesam) [inline x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:156]: Have [fun](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-10900011.1)!

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+		)
+		use amd64 && libs+=( nvidia-ngx )
+	fi
+
+	local lib soname
+	for lib in "${libs[@]}"; do
+		[[ ${lib:0:3} != lib ]] && lib=lib${lib}.so.${PV}
+
+		# auto-detect soname and create appropriate symlinks
+		soname=$(scanelf -qF'%S#F' "${lib}") || die "Scanning ${lib} failed"
```
Comment (thesamesam) [inline x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:232]: I'd explicitly BDEPEND on pax-utils

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+	linux-mod_pkg_preinst
+
+	# set video group id based on live system (bug #491414)
+	local g=$(getent group video | cut -d: -f3)
+	[[ ${g} ]] || die "Failed to determine video group id"
+	sed "s/PACKAGE/${PF}/;s/VIDEOGID/${g}/" \
+		-i "${ED}"/etc/modprobe.d/nvidia.conf || die
+
+	# try to find driver mismatches using temporary supported-gpus.json
+	for g in $(grep -l 0x10de /sys/bus/pci/devices/*/vendor 2>/dev/null); do
```
Comment (thesamesam) [inline x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:350]: I suppose we could/should explicitly BDEPEND on grep too.

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild (hunk 2 above)
Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:232]:
```
BDEPEND="
    app-misc/pax-utils
```
Is it not already explicit enough? ;)

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild (hunk 1 above)
Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:156]: It didn't matter if they were empty or set to something else outside of src_install (it doesn't hardcode paths), but if really must I guess I could complexify the ebuild over this.

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild (hunk 3 above)
Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:350]: Ah yes, unlike sed/patch/find it's not guaranteed by pms during build.

Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:350]: Oh wait, actually this should be RDEPEND, this is pkg_preinst

Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:350]: Or rather IDEPEND if we had it... cross-compiling nvidia-drivers is already rather fickle either way.

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild (hunk 1 above)
Comment (thesamesam) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:156]: _grumbles_

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild (hunk 2 above)
Comment (thesamesam) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:232]: Oh no!

File: x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild (hunk 1 above)
Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild:156]: Alright, alright :) Decided to re-arrange for varied other reasons, so variable usage will be correct!

**Review by SoapGentoo — APPROVED** [review]

File: x11-drivers/nvidia-drivers/nvidia-drivers-390.141-r1.ebuild [hunk 4]
```diff
@@ -47,6 +47,7 @@ COMMON_DEPEND="
 	)"
 RDEPEND="
 	${COMMON_DEPEND}
+	sys-apps/grep
```
[the remark this answers is not in the document]
Comment (ionenwks) [reply x11-drivers/nvidia-drivers/nvidia-drivers-390.141-r1.ebuild]: It's not, and after discussions it's been removed. Adding deps for sake of being pedantic in a non-pedantic way (wrog scope) sound kind of wrong. Maybe will revisit when we have CBUILD-IDEPEND if there's some stance on how to handle implicit base system deps on non-stage packages.

**Pull request closed without merging**